### PR TITLE
Add avx512 core instructions check

### PR DESCRIPTION
**Pull request by wozna**
### PR types

Bug fixes
### PR changes

OPs 
### Describe

This PR adds avx512 core instruction check in bfloat16 convolution test. Bfloat16 evaluation is possible only if the CPU has ` AVX512BW, AVX512VL, and AVX512DQ` flags. Without them, all tests with bfloat16 will fail.
It is related to https://github.com/PaddlePaddle/Paddle/issues/27588

### CODE REVIEW

File: paddle/fluid/pybind/pybind.cc [hunk 1]
```diff
@@ -142,6 +142,17 @@ bool IsCompiledWithMKLDNN() {
 #endif
 }
 
+bool IsBfloat16() {
```
Comment (wojtuss) [inline paddle/fluid/pybind/pybind.cc]: I would suggest renaming it into something like "SupportsBfloat16" or "CanRunBfloat16", as `IsBfloat16` does not convey the meaning of the function.

File: paddle/fluid/pybind/pybind.cc [hunk 2]
```diff
@@ -1661,6 +1672,7 @@ All parameter, weight, gradient are variables in Paddle.
   m.def("is_compiled_with_cuda", IsCompiledWithCUDA);
   m.def("is_compiled_with_xpu", IsCompiledWithXPU);
   m.def("is_compiled_with_mkldnn", IsCompiledWithMKLDNN);
+  m.def("is_bfloat16", IsBfloat16);
```
Comment (wojtuss) [inline paddle/fluid/pybind/pybind.cc]: ditto

File: paddle/fluid/pybind/pybind.cc (hunk 1 above)
Comment (wozna) [reply paddle/fluid/pybind/pybind.cc]: Yeah, you are right, I like `SupportsBfloat16`  :)

**Review by wojtuss — COMMENTED** [review]
LGTM

**Review by jczaja — APPROVED** [review]
LGTM

**Review by wojtuss — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@luotao1 Could you please start your review?

**Pull request merged**